api/v1alpha1/release_types_test.go

```diff
@@ -0,0 +1,446 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type commonArgs struct {
+	status  metav1.ConditionStatus
+	reason  ReleaseReason
+	message string
+}
+
+var _ = Describe("Release type", func() {
+
+	var r *Release
+	JustBeforeEach(func() {
```

Comment:
If you are not intended to use `BeforeEach` inside the `Context` blocks as well you don't need `JustBeforeEach`. Unless you are planning to do this later on.

Reply:
The decision to use `JustBeforeEach` was on purpose, to imperatively set things as close as possible to `It()` nodes.
It could be changed to `BeforeEach` if you feel strongly against `JustBeforeEach`? There was no plan to use `BeforeEach`.

That being said, I could be persuaded to set `Release` structures at the `Context` scope, and use `JustBeforeEach` to restore the structure fields prior to each test. That way we optimize for memory, in case somebody ever wants to run a benchmark or similar stress test, the runtime wouldn't thrash GC so much. To use a cliche, I'm not sure _the wick is worth the game_, but I'm willing to burn that candle if we want? What do you think?

Reply:
Moved to `BeforeEach`. Thanks. :+1: